projects/feature-activation/0002-bit-signaling.md

```diff
@@ -0,0 +1,64 @@
+
+Users running the full node, or specifically miners in this case, should be able to pass two new options to the `run_node` CLI interface:
+
+1. `--signal-support [feature]`, and
+2. `--signal-not-support [feature]`
+
+Here are rule definitions for these options:
+
+1. Both options receive the name of some feature, as defined in the existing `Feature` enum;
+2. Both options can be used multiple times to either enable or disable support for multiple features;
+3. A feature cannot be set in both options at the same time, as this would be a conflicting configuration;
+4. A feature can only be set in any of these options if it's currently undergoing a signaling phase of its Feature Activation process, otherwise setting its bit could conflict with other features;
+5. A default bit signal value (either enabled or disabled) should be provided for each feature.
+
+Here's an example of running the full node with those options, enabling support for features 2 and 3, and disabling support for feature 1:
+
+```bash
+poetry run hathor-cli run_node --status 8080 --testnet --memory-storage --signal-not-support NOP_FEATURE_1 --signal-support NOP_FEATURE_2 --signal-support NOP_FEATURE_3
+```
+
+Let's assume all features are disabled by default, and that `NOP_FEATURE_1` is configured to use `bit=0`, `NOP_FEATURE_2` is configured to use `bit=1`, and `NOP_FEATURE_3` is configured to use `bit=2`, where `bit=0` is the LSB, as explained in the previous RFC.
+
+Then, the signal bits for this full node run would be `110`, meaning that those bits would be present in all block templates generated for mining.
+
+# Reference-level explanation
+[Reference-level explanation]: #reference-level-explanation
+
+## Bit Signaling Serivce
+
+A new `BitSignalingService` class should be implemented, providing a single public method: `get_signal_bits()`, that returns an `int`. This class and method's only purpose is to calculate the binary number that represents the signal bits for all block templates generated during this full node run, as configured by the respective CLI options and feature configuration.
+
+That class should also perform validations on the configuration, as specified in the list in the [Guide-level explanation] section. The full node must not start if those validations fail.
+
+## New signal_support_by_default configuration
+
+A new attribute called `signal_support_by_default` should be implemented in the `Criteria` class. It is a boolean that represents whether a feature should be supported by default or not, according to its configuration via `HathorSettings`. Its default value should be `False`.
+
+That is, assuming `signal_support_by_default=True` for `NOP_FEATURE_1`, and no CLI option is provided for that feature, then that feature is going to be signaled as supported in block templates. If some CLI option is set for that feature, then it takes precedence over the `signal_support_by_default` attribute.
+
+## Hathor Manager and mining APIs
```

Comment:
All block template generators must also be updated accordingly.

Reply:
I added that line in https://github.com/HathorNetwork/rfcs/pull/59/commits/c0ec7e86771268e26cb2adf63bc51a31942fd3f1, but during implementation I'll have to reach out to you and/or Jan for specifics, to make sure we cover all template APIs

Reply:
Ok. We can do it at design time as well. I think every mining API uses the same block template generator. But we have to confirm it.

Reply:
Sure! In the POC I changed the `BlockTemplate(NamedTuple)` to add a new `signal_bits` attribute. This reflected in a change in `HathorManager`, that calls the new `BitSignalingService.get_signal_bits()` to set them in the `BlockTemplate`. But I'll check if that covers all mining APIs.

Reply:
I checked it, and doing it like it's done in the POC is enough to cover all mining APIs. Updated the text in https://github.com/HathorNetwork/rfcs/pull/59/commits/674770f6c47e8ed42a3bea70d72e4d9c4fd84fd3

Reply:
Good job! Thanks!